doc/decisions/0012-repeating-answer-options.md

```diff
@@ -0,0 +1,259 @@
+# 12. Dynamically creating answers based on a list
+
+## Context
+
+We need the ability to dynamically generate answers based on a list item.
+
+We need to support dynamic answers that are driven by:
+
+- List collectors
+- Arbitrary functions that return a sequence of values
+- Checkbox answers
+
+Additional requirements:
+
+- Dynamic answers need to be identifiable for downstream processing
+- Dynamic answers need to be able to support user defined answer codes
+
+## Proposal
+
+Introduce a new `dynamic_answers` object that will handle the generation of any repeating answers based on a list of items.
+
+## Properties and uses:
+
+### `dynamic_answers`
+
+- A new property within the `question` block that supports answers based on a list of items, answers, or arbitrary functions 
+  that return a sequence of values.
+- `dynamic_answers` will contain two top level properties, `values` and `answers`.
+- `dynamic_answers` can be used alongside the existing static `answers` property, allowing for static and dynamic answers 
+  on the same page.
+- `dynamic_answers` must have at least one but can contain multiple `answers` to be iterated over for each literal list item.
+- `dynamic_answers` will be limited to a single block i.e. they will only be displayed on one page.
+  
+#### `dynamic_answers.values`
+
+- This field will contain the `source` for the list of literal values to be repeated over.
+- The `values` property will follow existing patterns and allow either a `Value Source` or a `Value Operator` as per the schema examples below.
+- There must be at least one item in the list being used to generate dynamic answers.
```

Comment:
What happens if the list is empty?

Reply:
This is on the Author guys to prevent with routing rules. It will fall over if there is dynamic_answers and an  empty list